### PR TITLE
1804454: collect uuid on aarch64 system

### PR DESCRIPTION
**Pull request by wottop**

### CODE REVIEW

**Comment by martinpitt** [comment]
I'm triggering a rhel-8-3 test here. I think it makes sense to switch the cockpit CI from 8.2 to 8.3 now, and if that test succeeds, I'll flip it over for all PRs. If you'd rather still keep 8.2 (and thus run both 8.2 and 8.3), please yell.

**Comment by jirihnidek** [comment]
@martinpitt I think that it's ok to switch to rhel8.3

File: test/rhsmlib_test/test_collector.py [hunk 1]
```diff
@@ -43,3 +44,16 @@ def test_get_arch(self):
     def test_get_platform_specific_info_provider(self):
         info_provider = firmware_info.get_firmware_collector(arch=platform.machine())
         self.assertTrue(info_provider is not None)
+
+
+class GetNonDmiUuid(unittest.TestCase):
+    def test_get_aarch64_firmware_collector(self):
+        firmware_provider_class = firmware_info.get_firmware_collector(arch='aarch64')
+        self.assertTrue(isinstance(firmware_provider_class, UuidFirmwareInfoCollector))
```
Comment (jirihnidek) [inline test/rhsmlib_test/test_collector.py:52]: I would add one more unit test for the case the `/sys/devices/virtual/dmi/id/product_uuid` is not available or it is not possible to read it due to access permissions.

File: test/rhsmlib_test/test_collector.py [hunk 2]
```diff
@@ -43,3 +44,16 @@ def test_get_arch(self):
...
+
+    @mock.patch(OPEN_FUNCTION, mock.mock_open(read_data="356B6CCC-30C4-11B2-A85C-BBB0CCD29F36"))
+    def test_get_aarch64_uuid_collection(self):
+        firmware_provider_class = firmware_info.get_firmware_collector(arch='aarch64')
+        firmware_provider_class.arch = 'aarch64'
+        result = firmware_provider_class.get_all()
+        self.assertTrue(result['dmi.system.uuid'] == '356B6CCC-30C4-11B2-A85C-BBB0CCD29F36')
```
Comment (jirihnidek) [inline test/rhsmlib_test/test_collector.py:59]: BTW: I can see that the UUID is also visible as `virt.uuid`. So I would add one more assertion:

```python
        self.assertTrue(result['virt.uuid'] == '356B6CCC-30C4-11B2-A85C-BBB0CCD29F36')
```

**Pull request merged**